src/components/internal/InternalRadioGroup.tsx

```diff
@@ -164,15 +207,26 @@ function InternalRadioGroup(_props) {
   const cols = columns === undefined ? options.length : columns;
   const areLabelsBold = options.some((option) => option.description);
 
+  const RadioGroupRef = useRef<HTMLDivElement>(null);
+
+  const onRadioBlur: React.FocusEventHandler<HTMLInputElement> = (event) => {
+    if (
+      !RadioGroupRef.current?.contains(event.relatedTarget as HTMLInputElement)
+    ) {
+      onBlur?.(event);
+    }
+  };
   return (
     <div
       role="radiogroup"
-      aria-invalid={isValid ? null : "true"}
+      ref={mergeRefs([RadioGroupRef, innerRef ?? null])}
+      data-testid={testId}
```

Comment:
again, I think we shouldn't need testIds on internal components.

Reply:
see comment above